sfge/src/main/java/com/salesforce/rules/UnusedMethodRule.java

```diff
@@ -161,84 +200,204 @@ private boolean methodIsIneligible(MethodVertex vertex) {
+    /**
+     * @return True if this provided method is visible to subclasses of the class where it's
+     *     defined.
+     */
+    private boolean methodVisibleToSubclasses(MethodVertex methodVertex) {
+        // If a method is private, it's not visible to subclasses.
+        if (methodVertex.isPrivate()) {
+            return false;
+        }
+        // If a method is protected or public, then it's visible to subclasses.
+        // But subclasses can only exist if the parent class is abstract or virtual.
+        UserClassVertex classVertex =
+                methodVertex
+                        .getParentClass()
+                        .orElseThrow(() -> new UnexpectedException(methodVertex));
+        return classVertex.isAbstract() || classVertex.isVirtual();
+    }
+
+    /**
+     * Seeks invocations of the provided method that occur within subclasses of the class where it's
+     * defined. E.g., 'this.method()` if not overridden, or `super.method()` if overridden.
+     *
+     * @return - True if such an invocation is found, else false.
+     */
+    private boolean methodUsedBySubclass(MethodVertex methodVertex) {
+        // Instantiate a visitor to use for checking subclass calls.
+        SubclassCallValidator visitor = new SubclassCallValidator(methodVertex);
+        // Get a list of each subclass of the method's host class.
+        List<String> subclasses = getSubclasses(methodVertex.getDefiningType());
+        // We'll want a loop so we can recursively process subclasses if needed.
+        while (!subclasses.isEmpty()) {
+            // For each of the subclasses...
+            for (String subclass : subclasses) {
+                // Get every method call in that class.
+                List<InvocableWithParametersVertex> potentialSubclassCallers =
+                        getMethodCalls(subclass);
+                // If the validator detects a usage in the subclass, then we're set.
+                if (validatorDetectsUsage(visitor, potentialSubclassCallers)) {
+                    return true;
+                }
+            }
+            // If we're here, then we've checked every subclass at this level.
+            // For a constructor, we're done, since constructors are only
+            // visible to immediate children.
+            // But for other methods, nested calls are possible, so we should
+            // get every subclass of the subclasses and keep going.
+            if (methodVertex.isConstructor()) {
+                break;
+            } else {
+                subclasses = getSubclasses(subclasses.toArray(new String[] {}));
             }
-            // If we're here, then our checks were satisfied, and this method call
-            // appears to be an invocation of the target method.
-            return true;
         }
-        // If we're at this point, none of the method calls satisfied our checks.
+        // If we're here, then we've fully examined the subclasses and found nothing.
```

Comment:
Not for this iteration, but I think this could be optimized so that we examine the subclasses for each extensible/visible method in the current class in one go.